lms/djangoapps/courseware/utils.py

```diff
@@ -95,3 +102,100 @@ def can_show_verified_upgrade(user, enrollment, course=None):
 
     # Show the summary if user enrollment is in which allow user to upsell
     return enrollment.is_active and enrollment.mode in CourseMode.UPSELL_TO_VERIFIED_MODES
+
+
+def is_eligible_for_financial_aid(course_id):
+    """
+    Sends a get request to edx-financial-assistance to retrieve financial assistance eligibility criteria for a course.
+
+    Returns either True if course is eligible for financial aid and vice versa.
+    Also returns the reason why the course isn't eligible.
+    In case of a bad request, returns an error message.
+    """
+    is_eligible_url = '/core/api/course_eligibility/'
+    financial_assistance_configuration = FinancialAssistanceConfiguration.current()
+    if financial_assistance_configuration.enabled:
+        oauth_application = Application.objects.get(user=financial_assistance_configuration.get_service_user())
+        client = OAuthAPIClient(
+            settings.LMS_ROOT_URL,
+            oauth_application.client_id,
+            oauth_application.client_secret
+        )
+        response = client.request('GET', f"{financial_assistance_configuration.api_url}{is_eligible_url}{course_id}/")
+        if response.status_code == status.HTTP_200_OK:
+            return response.json().get('is_eligible'), response.json().get('reason')
+        elif response.status_code == status.HTTP_400_BAD_REQUEST:
+            return False, response.json().get('message')
+    else:
+        return False, 'Financial Assistance configuration is not enabled'
+
+
+def get_financial_assistance_application_status(user_id, course_id):
+    """
+    Given the course_id, sends a get request to edx-financial-assistance to retrieve
+    financial assistance application status for the logged-in user.
+    """
+    application_status_url = f"/core/api/financial_assistance_application/status/?" \
+                             f"{parse.urlencode({'course_id': course_id})}&{parse.urlencode({'lms_user_id': user_id})}"
+    financial_assistance_configuration = FinancialAssistanceConfiguration.current()
+    if financial_assistance_configuration.enabled:
+        oauth_application = Application.objects.get(user=financial_assistance_configuration.get_service_user())
+        client = OAuthAPIClient(
+            settings.LMS_ROOT_URL,
+            oauth_application.client_id,
+            oauth_application.client_secret
+        )
```

Comment:
these 3 lines are duplicated across all utils methods. Maybe create a util method to get the client or None if Faid is not enabled.